modules/social_features/social_user/social_user.module

```diff
@@ -801,3 +822,43 @@ function social_user_module_implements_alter(&$implementations, $hook) {
     $implementations['social_user'] = $social_user;
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ *
+ * @throws \Drupal\Core\Entity\EntityStorageException
+ */
+function social_user_user_insert(UserInterface $user) {
+  // If site setting "Registered users are verified immediately" is enabled:
+  // new registered users get the "verified user & authenticated" role when they
+  // are registered.
+  $config = \Drupal::config('social_user.settings');
+  $verified_immediately = $config->get('verified_immediately');
+  if ($verified_immediately) {
+    $user->addRole('verified');
+    $user->save();
+  }
+}
+
+/**
+ * Implements hook_select2_autocomplete_matches_alter().
+ */
+function social_user_select2_autocomplete_matches_alter(array &$matches, array $options) {
```

Comment:
Shouldn't we tackle this at the original query? Not sure if that's possible though.

Loading in accounts, checking if they are verified and if not unsetting them is a possible performance issue and also can potentially lead to empty lists of autocompletes.

Reply:
I will check this option.

Reply:
I'm afraid we cannot tackle this at the original query. But I have moved this logic from `hook_select2_autocomplete_matches_alter()` to [Select2EntityAutocompleteMatcher.php](https://github.com/goalgorilla/open_social/blob/10.3.x/modules/social_features/social_core/src/Entity/Select2EntityAutocompleteMatcher.php) where we are altering `select2.autocomplete_matcher` service.